lib/phlex/sgml.rb

```diff
@@ -223,6 +223,19 @@ def render(renderable = nil, &)
 		nil
 	end
 
+	def cache(*keys, **, &)
+		context = @_context
+		return if context.fragments && !context.in_target_fragment
+
+		context.buffer << cache_store.fetch(
+			[Phlex::DEPLOY_KEY, self.class.name, __method__, keys].freeze, **
+		) { capture(&) }
+	end
+
+	def cache_store
+		Phlex::CACHE_STORE
+	end
```

Comment:
We go via the `cache_store` method so you can use different cache stores for different components.